applier: fix subscribe_request::id_filter initialization (#11060)
tarantool/tarantool: changelogs/unreleased/gh-10592-replica-does-not-receive-its-own-rows.md

## bugfix/replication

> * Fixed a bug when a master who crashed and lost his xlogs could not get his
>   own rows when it tried to re-join the cluster and hangs forever (gh-10592).

Reviewer: ```suggestion
* Fixed a bug when a master that crashed and lost its xlogs could not get its
  own rows when it tried to re-join the cluster and hanged forever (gh-10592).
```

Author: Applied.
